lib/streamlit/elements/media.py

```diff
@@ -292,12 +324,41 @@ def marshall_video(
         if youtube_url := _reshape_youtube_url(data):
             proto.url = youtube_url
             proto.type = VideoProto.Type.YOUTUBE_IFRAME
+            if subtitles:
+                raise StreamlitAPIException(
+                    "Subtitles are not supported for YouTube videos."
+                )
         else:
             proto.url = data
 
     else:
         _marshall_av_media(coordinates, proto, data, mimetype)
 
+    if subtitles:
+        subtitle_items = []
+
+        # Single subtitle
+        if isinstance(subtitles, (str, bytes, io.BytesIO)):
+            subtitle_items.append(("default", subtitles))
+        # Multiple subtitles
+        elif isinstance(subtitles, dict):
+            subtitle_items.extend(subtitles.items())
+        else:
+            raise StreamlitAPIException(
+                f"Unsupported data type for subtitles: {type(subtitles)}. "
+                f"Only str (file paths) and dict are supported."
+            )
+
+        for label, subtitle_data in subtitle_items:
+            sub = proto.subtitles.add()
+            sub.label = label
```

Comment:
@kajarenc One detail not called out clearly in the spec is that we should support the case where users pass in subtitle files **but** don't want to display any tracks by default. E.g

```py
st.video("example.mp4", subtitles={"":"", "Deutsch": "subtitles-de.vtt", "English": "subtitles-en.srt"})
```

We should support empty strings for both label and URL.

```suggestion
            sub.label = label if label else ""
```